src/uct/api/uct.h

```diff
@@ -1747,6 +1779,16 @@ ucs_status_t uct_ep_create(const uct_ep_params_t *params, uct_ep_h *ep_p);
  * @brief Initiate a synchronized disconnection of an endpoint connected to a
  *        sockaddr by a connection manager @ref uct_cm_h.
  *
+ * This routine will disconnect the given endpoint from its remote peer.
+ * The remote side, should also call this routine when handling the initiator's
+ * disconnect.
+ * After a call to this function, the given endpoint should not be used for
+ * communication anymore.
+ * @ref uct_ep_destroy needs to be called on this endpoint after the remote
+ * disconnect was handled by @ref uct_ep_params::disconnect_cb or after the
```

Comment:
adding @evgeny-leksikov 
why do we need to request to invoke the remote disconnect instead of the completion?
seems cleaner if we just use the completion everytime
disconnect_cb can be called, if uct_ep_disconnect was not called before

Reply:
User can pass NULL-completion to `uct_ep_disconnect`, I think we need to invoke `disconnect_cb` in this case. There are different arguments to disconnect_cb and UCT-completion. The last one requires extra infrastructure from user side. We do that in UCP to reduce EP memory footprint but other app may not have such strict requirements and use only disconnect_cb path. This is simpler.